Sample_v3/Samples/CombineSimpleChat/CombineSimpleChatViewController.swift

```diff
@@ -0,0 +1,437 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import StreamChatClient
+import UIKit
+
+///
+/// # CombineSimpleChatViewController
+///
+/// A `UITableViewController` subclass that displays and manages a channel.  It uses the `ChannelController`  class to make calls to the Stream Chat API and listens to
+/// events by conforming to `ChannelControllerDelegate`.
+///
+@available(iOS 13, *)
+final class CombineSimpleChatViewController: UITableViewController {
+    // MARK: - Properties
+    
+    ///
+    /// # channelController
+    ///
+    ///  The property below holds the `ChannelController` object.  It is used to make calls to the Stream Chat API and to listen to the events. After it is set,
+    ///  we need to start observing `ChannelController` event.
+    ///  While using Combine we should subscribe to `Publishers` events.
+    var channelController: ChannelController! {
+        didSet {
+            subscribeToCombinePublishers()
+        }
+    }
+    
+    // MARK: - Combine
+
+    private lazy var cancellables: Set<AnyCancellable> = []
+    
+    ///
+    /// # subscribeToCombinePublishers
+    ///
+    /// Here we bind `channelControllers` publishers so we can observe the changes.
+    private func subscribeToCombinePublishers() {
+        /// `ChannelController` will not trigger the `channelChangePublisher` on the initial channel set so
+        /// we can` prepend` our `channelChangePublisher` sequence with the initial channel manually.
+        let initialChannel = Just(channelController.channel)
+            .compactMap { $0 }
+            .map { EntityChange<Channel>.update($0) }
+        
+        /// This subscription updates the view controller's `title` and its `navigationItem.prompt` to display the count of channel
+        /// members and the count of online members. When the channel is deleted, this view controller is dismissed.
+        let updatedChannel = channelController
+            .channelChangePublisher
+            /// Update UI for initial channel.
+            .prepend(initialChannel)
+            /// Dismiss VC and break the sequence if channel got deleted.
+            /// Map `EntityChange` to `Channel` and continue executing sequence if it is `update` change.
+            .compactMap { [weak self] change -> Channel? in
+                switch change {
+                case .create:
+                    return nil
+                case let .update(channel):
+                    return channel
+                case .remove:
+                    self?.dismiss(animated: true)
+                    return nil
+                }
+            }
+            .share()
+        
+        updatedChannel
+            .map { $0.extraData.name ?? $0.cid.description }
+            .receive(on: RunLoop.main)
```

Comment:
Maybe `updatedChannel` can be received on main loop so you don't have to write it every time?